File: gene_Adios2_cori/external/futils-gene-mod/examples/sim.m
%
%     Get data from data sets
%
time=hdf5read('sim.h5','/scalars/time');
epot=hdf5read('sim.h5','/scalars/epot');
ekin=hdf5read('sim.h5','/scalars/ekin');
%
%     Attributes
%
attr=hdf5read('sim.h5','/title'); title_ann=attr.Data;
attr=hdf5read('sim.h5','/scalars/time/title'); time_ann=attr.Data;
attr=hdf5read('sim.h5','/scalars/epot/title'); epot_ann=attr.Data;
attr=hdf5read('sim.h5','/scalars/ekin/title'); ekin_ann=attr.Data;

figure
plot(time, ekin, time, epot)
legend(epot_ann, ekin_ann)
xlabel(time_ann)
title(title_ann)
grid on

File: gene_adios2/external/futils-gene-mod/examples/sim.py
#!/usr/bin/env python

from tables import *
from pylab import *

f=openFile("sim.h5", mode="r")
timeid=f.getNode('/scalars','time')
epotid=f.getNode('/scalars','epot')
ekinid=f.getNode('/scalars','ekin')

print "Number of steps = %d"% (timeid.nrows,)

time_ann  = timeid.getAttr('title')
epot_ann = epotid.getAttr('title')
ekin_ann = ekinid.getAttr('title')
title_ann = f.getNodeAttr('/', 'title')

time=timeid[:]
epot=epotid[:]
ekin=ekinid[:]

f.close()

hold(True)
plot(time, epot, 'b', label=epot_ann)
plot(time, ekin, 'r', label=ekin_ann)
legend(loc='best')
xlabel(time_ann);
title(title_ann);
grid(True)

show()
